src/SuperSelectField.js

```diff
@@ -406,10 +429,35 @@ class SelectField extends Component {
     event.preventDefault()
     const { selectedItems } = this.state
     if (this.props.multiple) {
-      const selectedItemExists = selectedItems.some(obj => areEqual(obj.value, selectedItem.value))
-      const updatedValues = selectedItemExists
-        ? selectedItems.filter(obj => !areEqual(obj.value, selectedItem.value))
-        : selectedItems.concat(selectedItem)
+      const { allowSelectAll, selectAllItem } = this.props;
+      let updatedValues;
+      if (allowSelectAll && selectedItem.value === selectAllItem.value) {
```

Comment:
It is an object of the format `{ value: 'all-superSelectField', label: 'All' }`. It is in the PropTypes for the select field. I figured it _could_ be possible someone might already have an existing value that is `all`, so wanted to give them the ability to change it if necessary.

Is there a better name for it?